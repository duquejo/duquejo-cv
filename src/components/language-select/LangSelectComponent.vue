
<template>
  <select 
    v-if="availableLocales" 
    :id="props.id" 
    class="text-gray-600 lg:text-gray-800 bg-transparent text-base lg:appearance-none lg:w-10/12 text-center outline-gray-300 cursor-pointer capitalize"
    v-model="locale"
    :aria-label="t('general.menu.language_selector_label')">
    <option
      v-for="aLocale in availableLocales"
      :key="`locale-${aLocale}`"
      :value="aLocale">
      {{ aLocale }}
    </option>
  </select>
</template>

<script setup lang="ts">
import { useI18n } from 'vue-i18n';

const { t, locale, availableLocales } = useI18n();

interface Props {
  id: string;
}

const props = withDefaults(defineProps<Props>(),{
  id: 'language-selector'
});
</script>