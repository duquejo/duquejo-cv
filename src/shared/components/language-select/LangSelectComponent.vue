
<template>
  <select class="text-gray-500 lg:text-gray-800 bg-transparent text-base lg:appearance-none lg:w-10/12 text-center outline-gray-300 cursor-pointer" v-model="locale">
    <option selected value="es-CO">ES</option>
    <option value="en-US">EN</option>
  </select>
</template>

<script setup lang="ts">
import { useI18n } from 'vue-i18n';
const { locale } = useI18n({ useScope: 'global' });
</script>