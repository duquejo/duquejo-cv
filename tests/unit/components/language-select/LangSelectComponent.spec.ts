import { describe, expect, it, vi } from 'vitest';
import { VueWrapper, mount } from '@vue/test-utils';
import createVueI18n from '../../../mocks/mock-i18n';
import LangSelectComponent from '../../../../src/components/language-select/LangSelectComponent.vue';

describe('+ LangSelectComponent.vue tests', () => {

  const i18n = createVueI18n();

  let wrapper: VueWrapper;
  wrapper = mount(LangSelectComponent, {
    global: {
      plugins: [i18n],
    },
  });

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findComponent(LangSelectComponent)).toBeTruthy();
  });

  it('Should append the available classes', () => {
    expect(wrapper.classes()).toEqual(['text-gray-500', 'lg:text-gray-800', 'bg-transparent', 'text-base', 'lg:appearance-none', 'lg:w-10/12', 'text-center', 'outline-gray-300', 'cursor-pointer', 'capitalize' ]);
  });

  it('Should render a select element with the whole available locales', () => {
    const select = wrapper.find('select');
    expect(select.exists()).toBeTruthy();
    
    const options = select.findAll('option');
    expect(options).toHaveLength(3);

    expect(options[0].text()).toBe('en');
    expect(options[1].text()).toBe('es');
    expect(options[2].text()).toBe('fr');
  });

  it('should update the locate status if a specified option is selected', async () => {
    const newValue = 'fr';
    const select = wrapper.find('select');
    await select.setValue(newValue);
    expect(select.element.value).toBe(newValue);
  });
});