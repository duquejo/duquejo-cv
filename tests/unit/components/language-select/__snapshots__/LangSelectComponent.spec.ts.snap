// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`+ LangSelectComponent.vue tests > Should match the snapshot 1`] = `
"<select id="foo-bar" class="font-semibold text-gray-600 lg:text-gray-800 bg-transparent text-base lg:appearance-none lg:w-10/12 text-center outline-gray-300 cursor-pointer capitalize" aria-label="general.menu.language_selector_label">
  <option value="en">en</option>
  <option value="es">es</option>
  <option value="fr">fr</option>
</select>"
`;
